rasa_core/training/interactive.py

```diff
@@ -792,14 +805,60 @@ def _validate_action(action_name,  # type: Text
     if not answers["action"]:
-        corrected_action = _request_action_from_user(predictions, sender_id,
-                                                     endpoint)
-        _correct_wrong_action(corrected_action, endpoint, sender_id,
+        action_name = _request_action_from_user(predictions, sender_id,
+                                                endpoint)
+
+    tracker = retrieve_tracker(endpoint, sender_id,
+                               EventVerbosity.AFTER_RESTART)
+
+    if (tracker.get('active_form', {}).get('name') and
+            action_name not in {tracker['active_form']['name'],
+                                ACTION_LISTEN_NAME}):
+        # notify the tracker that form was rejected
+        send_event(endpoint, sender_id,
+                   {"event": "action_execution_rejected",
+                    "name": tracker['active_form']['name']})
+    elif (action_name == tracker.get('active_form', {}).get('name') and
+            tracker.get('active_form', {}).get('rejected')):
+        # active form was chosen after it was rejected
+        # ask a user whether an input should be validated
+        q = ("Should '{}' validate user input to fill the slot '{}'?"
+             "".format(action_name, tracker.get("slots",
+                                                {}).get(REQUESTED_SLOT)))
+        validation_questions = [{
+            "name": "validation",
+            "type": "confirm",
+            "message": q
+        }]
+        form_answers = _ask_questions(validation_questions, sender_id,
+                                      endpoint)
+        if not form_answers["validation"]:
+            # notify form action to skip validation
+            send_event(endpoint, sender_id, {"event": "form_validation",
+                                             "validate": False})
+        elif not tracker.get('active_form', {}).get('validate'):
+            # handle contradiction with learned behaviour
+            q = ("WARNING: FormPolicy predicted no form validation "
+                 "based on previous training stories. "
+                 "Make sure to remove contradictory stories "
+                 "from training data.")
+            warning_questions = [{
+                "name": "warning",
+                "type": "input",
+                "message": q
+            }]
+            _ask_questions(warning_questions, sender_id, endpoint)
+            # notify form action to validate an input
+            send_event(endpoint, sender_id, {"event": "form_validation",
+                                             "validate": True})
+
+    if not answers["action"]:
+        _correct_wrong_action(action_name, endpoint, sender_id,
                               finetune=finetune)
-        return corrected_action == ACTION_LISTEN_NAME
     else:
-        send_action(endpoint, sender_id, action_name)
-        return action_name == ACTION_LISTEN_NAME
+        send_action(endpoint, sender_id, action_name, policy, confidence)
```

Comment:
i guess you've tested all this interactive stuff to make sure we haven't broken normal interactive learning?

Reply:
well... Might make sense to test it more thoroughly

Reply:
yes definitely. i might have some time today